### PR TITLE
Clarify functions and variables names

### PR DESCRIPTION
**Pull request by alexblnn**
This PR is a first attempt at clarifying the code, and addressing #18.

### CODE REVIEW

File: sanssouci/lambda_calibration.py [hunk 1]
```diff
@@ -50,30 +50,29 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
...
     # Init
-    n = X.shape[0]
-    p = X.shape[1]
+    n, p = X.shape
 
     # Step 1: calculate $p$-values for B permutations of the class assignments
 
     # 1.1: Intialise all vectors and matrices
-    shuffled_categ_current = categ.copy()
+    labels_shuffled = labels.copy()
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:58]:
```suggestion
    shuffled_labels = labels.copy()
```

File: sanssouci/lambda_calibration.py [hunk 2]
```diff
@@ -50,30 +50,29 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
...
 
-    shuffled_categ_all = np.zeros([B, n])
+    labels_shuffled_all = np.zeros((B, n))
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:60]:
```suggestion
    all_shuffled_labels = np.zeros((B, n))
```

File: sanssouci/post_hoc_bounds.py [hunk 3]
```diff
@@ -38,33 +38,34 @@ def max_fp(p_values, thr):
         Annals of Statistics, 48(3), 1281-1303.
     """
 
-    # make sure that thr is sorted
-    if np.linalg.norm(thr - thr[np.argsort(thr)]) > 0.0001:
-        thr = thr[np.argsort(thr)]
-        print("The input thr was not sorted -> this is done now")
+    # make sure that thresholds is sorted
+    if np.linalg.norm(thresholds - thresholds[np.argsort(thresholds)]) \
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:42]:
```suggestion
    if np.linalg.norm(thresholds - np.sort(thresholds)) > .0001:
```

File: sanssouci/post_hoc_bounds.py [hunk 4]
```diff
@@ -38,33 +38,34 @@ def max_fp(p_values, thr):
...
+       > 0.0001:
+        thresholds = thresholds[np.argsort(thresholds)]
+        print("The input 'thresholds' was not sorted -> this is done now")
 
     # do the job
-    nS = p_values.shape[0]
-    K = thr.shape[0]
-    size = np.min([nS, K])
+    subset_size = p_values.shape[0]
+    K = thresholds.shape[0]
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:49]: Can you call that `n_tests` (?) instead of `K` ?

File: sanssouci/post_hoc_bounds.py [hunk 5]
```diff
@@ -154,22 +155,24 @@ def curve_max_fp(p_values, thr):
...
-    #  make sure that p_values and thr are sorted
+    #  make sure that p_values and thresholds are sorted
     if np.linalg.norm(p_values - p_values[np.argsort(p_values)]) > 0.0001:
         p_values = p_values[np.argsort(p_values)]
         print("The input p-values were not sorted -> this is done now")
 
-    if np.linalg.norm(thr - thr[np.argsort(thr)]) > 0.0001:
-        thr = thr[np.argsort(thr)]
-        print("The input thr were not sorted -> this is done now")
+    if np.linalg.norm(thresholds - thresholds[np.argsort(thresholds)]) \
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:163]: see above

File: sanssouci/post_hoc_bounds.py [hunk 6]
```diff
@@ -154,22 +155,24 @@ def curve_max_fp(p_values, thr):
...
+       > 0.0001:
+        thresholds = thresholds[np.argsort(thresholds)]
+        print("The input 'thresholds' were not sorted -> this is done now")
 
     # do the job
     p = p_values.shape[0]
-    kMax = thr.shape[0]
+    kMax = thresholds.shape[0]
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:170]:
```suggestion
    k_max = thresholds.shape[0]
```

File: sanssouci/row_welch.py [hunk 7]
```diff
@@ -33,54 +33,54 @@ def get_summary_stats(mat, categ):
...
-        sumc = np.sum(matc, axis=0)
-        sum2c = np.sum(matc * matc, axis=0)
-        nc = ww.shape[0]
-        mc = sumc / nc
-        sc = np.sqrt((sum2c - ((sumc * sumc) / nc)) / (nc - 1))
-
-        res[cc] = {"sum": sumc, "sum2": sum2c, "n": nc, "mean": mc, "sd": sc}
+    for lab in labels_set:
+        where_lab = np.where(labels == lab)[0]
+        X_lab = X[where_lab, :]
```
Comment (bthirion) [inline sanssouci/row_welch.py:42]:
```suggestion
        X_lab = X[labels == lab]
```

File: sanssouci/row_welch.py [hunk 8]
```diff
@@ -92,25 +92,26 @@ def suff_welch_test(mx, my, sx, sy, nx, ny):
...
 
     # approximate degrees of freedom (Welch-Satterthwaite)
-    df = sse2 / ((sse_x * sse_x / (nx - 1)) + ((sse_y * sse_y) / (ny - 1)))
+    df = squared_sem * squared_sem /\
+        ((squared_sem_x * squared_sem_x / (n_x - 1)) +
+         ((squared_sem_y * squared_sem_y) / (n_y - 1)))
 
-    # p-value
-    pval = 2 * (1 - stats.t.cdf(np.abs(stat), df=df))
+    # p-value (two-sided!)
```
Comment (bthirion) [inline sanssouci/row_welch.py:108]:
```suggestion
    # two-sided p-value
```

File: sanssouci/row_welch.py [hunk 9]
```diff
@@ -150,16 +151,13 @@ def row_welch_tests(mat, categ):
 
     """
 
-    # n = mat.shape[0]
-    # p = mat.shape[1]
-
-    sstats = get_summary_stats(mat, categ)
+    sstats = get_summary_stats(X, labels)
```
Comment (bthirion) [inline sanssouci/row_welch.py:154]:
```suggestion
    summary_stats = get_summary_stats(X, labels)
```

**Review by bthirion — COMMENTED** [review]
That's a great imprvement !

File: sanssouci/lambda_calibration.py (hunk 1 above)
Comment (alexblnn) [reply sanssouci/lambda_calibration.py]: fixed in c2fd1a3ed906d58ba57807a78aa25e7237ccd87e

File: sanssouci/lambda_calibration.py (hunk 2 above)
Comment (alexblnn) [reply sanssouci/lambda_calibration.py]: fixed in c2fd1a3ed906d58ba57807a78aa25e7237ccd87e

File: sanssouci/lambda_calibration.py [hunk 10]
```diff
@@ -36,7 +36,7 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
     Returns
     -------
 
-    pva0 : array-like of shape (B, p)
+    pval0 : array-like of shape (B, p)
```
Comment (pneuvial) [inline sanssouci/lambda_calibration.py:39]:
```suggestion
    null_p_values : array-like of shape (B, p)
```

File: sanssouci/post_hoc_bounds.py (hunk 4 above)
Comment (pneuvial) [reply sanssouci/post_hoc_bounds.py:49]: `K` is not the number of tests but the size of the template, so if we want an explicit name I'd suggest `template_size` or something alike.

File: sanssouci/row_welch.py [hunk 11]
```diff
@@ -33,54 +33,54 @@ def get_summary_stats(mat, categ):
...
-        matc = mat[ww, :]
-        sumc = np.sum(matc, axis=0)
-        sum2c = np.sum(matc * matc, axis=0)
-        nc = ww.shape[0]
-        mc = sumc / nc
-        sc = np.sqrt((sum2c - ((sumc * sumc) / nc)) / (nc - 1))
-
-        res[cc] = {"sum": sumc, "sum2": sum2c, "n": nc, "mean": mc, "sd": sc}
+    for lab in labels_set:
+        where_lab = np.where(labels == lab)[0]
```
Comment (pneuvial) [inline sanssouci/row_welch.py:41]:
```suggestion
```

File: sanssouci/row_welch.py [hunk 12]
```diff
@@ -33,54 +33,54 @@ def get_summary_stats(mat, categ):
...
+        X_lab = X[where_lab, :]
+        sum_lab = np.sum(X_lab, axis=0)
+        sum_squares_lab = np.sum(X_lab * X_lab, axis=0)
+        n_lab = where_lab.shape[0]
```
Comment (pneuvial) [inline sanssouci/row_welch.py:45]:
```suggestion
        n_lab = np.sum(labels == lab)
```

**Review by pneuvial — APPROVED** [review]
Looks great, thanks!

File: sanssouci/lambda_calibration.py [hunk 13]
```diff
@@ -50,30 +50,29 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
...
     pval0 = np.zeros([B, p])
 
     if row_test_fun == row_welch_tests:  # row Welch Tests (parallelized)
         for bb in range(B):
-            swt = row_welch_tests(X, shuffled_categ_all[bb, :])
-            pval0[bb, :] = swt['p_value'][:]
+            permuted_test_result = row_welch_tests(X, all_shuffled_labels[bb])
+            pval0[bb] = permuted_test_result['p_value']
     else:                     # standard scipy tests
         for bb in range(B):
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:73]:
```suggestion
        for b in range(B):
``` Not related to your changes, but would be more readable imho

File: sanssouci/post_hoc_bounds.py [hunk 14]
```diff
@@ -38,33 +38,34 @@ def max_fp(p_values, thr):
...
-    size = np.min([nS, K])
+    subset_size = p_values.shape[0]
+    template_size = thresholds.shape[0]
+    size = np.min([subset_size, template_size])
 
     if size < 1:
         return 0
 
-    seqK = np.arange(size)
+    seq_k = np.arange(size)
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:55]: The naming could still be improved: `seq_k` is both unclear and hard to pronounce.

File: sanssouci/post_hoc_bounds.py [hunk 15]
```diff
@@ -73,10 +74,10 @@ def min_tp(p_values, thr):
 
     p_values : 1D numpy.array
         A 1D numpy array of p-values for the selected items
-    thr : 1D numpy.array
+    thresholds : 1D numpy.array
         A 1D numpy array of non-decreasing k-FWER-controlling thresholds
 
-    Returns
+    Retursubset_size
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:80]: oops. Normally, such a line should trigger an error. If not, it means that the test suite is not complete.

File: sanssouci/post_hoc_bounds.py [hunk 16]
```diff
@@ -154,44 +155,46 @@ def curve_max_fp(p_values, thr):
...
+        if thresholds[kk] >= p_values[ii]:
             K[ii] = kk
             ii += 1
         else:
             Z[kk] = ii
             kk += 1
 
-    Vbar = np.zeros(p)
+    max_fp_ = np.zeros(p)
     ww = np.where(K > 0)[0]
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:196]: this `ww` variable is probably not useful, and can simply be replaced by `K > 0` wherever it appears.

**Review by bthirion — COMMENTED** [review]
Overall, this is a great PR, with lots of improvements. Please address pending comments so that we can merge it.

**Comment by alexblnn** [comment]
it looks good to me; we can merge if you are ok with this @bthirion @pneuvial

**Review by bthirion — APPROVED** [review]

**Comment by pneuvial** [comment]
Thanks a lot!

**Pull request merged**